src/bindings/js/node/src/model_wrap.cpp

```diff
@@ -128,3 +129,27 @@ Napi::Value ModelWrap::is_dynamic(const Napi::CallbackInfo& info) {
     const auto result = _model->is_dynamic();
     return Napi::Boolean::New(env, result);
 }
+
+Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
+    if (info.Length() != 1 || !info[0].IsNumber()) {
+        reportError(info.Env(), "Invalid argument. Expected a single number.");
+        return Napi::Value();
+    }
+
+    auto idx = info[0].As<Napi::Number>().Int32Value();
+    auto cm_outputs = _model->outputs();  // Output<Node>
+
+    if (idx < 0 || idx >= cm_outputs.size()) {
+        reportError(info.Env(), "Invalid index. Index out of range.");
+        return Napi::Value();
+    }
```

Comment:
Good check, but it is extra in this case. My proposals:
- use native https://docs.openvino.ai/2024/api/c_cpp_api/classov_1_1_model.html#_CPPv4NK2ov5Model16get_output_shapeE6size_t
- wrap into `try - catch` block (see examples in this file or in core.cpp)